packages/playwright-core/src/server/chromium/crBrowser.ts

```diff
@@ -471,6 +472,14 @@ export class CRBrowserContext extends BrowserContext {
 
   async _doClose() {
     assert(this._browserContextId);
+    const openedBeforeUnloadDialogs: Dialog[] = [];
```

Comment:
Let's comment that headed chrome does not close with beforeunload open, but not with any other dialog types, and we do not expect any new beforeunload dialogs because our close does not trigger them.